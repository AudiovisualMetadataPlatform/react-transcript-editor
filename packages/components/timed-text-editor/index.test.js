import React from 'react';
import { render, cleanup } from 'react-testing-library';
import TimeTextEditor from './index';

import bbcKaldiTranscript from './stories/fixtures/bbc-kaldi.json';

const mediaUrl = 'https://download.ted.com/talks/KateDarling_2018S-950k.mp4';

const defaultProps = {
  transcriptData: bbcKaldiTranscript,
  mediaUrl: mediaUrl,
  isEditable: true,
  spellCheck: false,
  onWordClick: () => {},
  sttJsonType: 'bbckaldi',
  isPlaying: () => {},
  playMedia: () => {},
  currentTime: 0,
  isScrollIntoViewOn: true,
  isPauseWhileTypingOn: true,
  timecodeOffset: 0,
  handleAnalyticsEvents: () => {},
  showSpeakers: true,
  showTimecodes: true,
  fileName: 'KateDarling_2018S-950k.mp4'
};

afterEach(cleanup);

describe('TimeTextEditor', () => {
  it('renders the editor correctly', () => {
    const { getByText } = render(<TimeTextEditor { ...defaultProps }/>);
    getByText('');
  });

  it('realigns the timecodes of text when text is written to');
  it('realigns the timecodes of text when text is deleted');
  it('pops up a modal when the speaker is selected');
  it('edits the speaker');
  it('underlines the time codes on hover');
  it('creates a new block of text when the "enter" is inserted');

});