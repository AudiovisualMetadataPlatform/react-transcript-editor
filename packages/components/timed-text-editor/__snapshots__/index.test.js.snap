// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TimeTextEditor /> should render correctly in "debug" mode 1`] = `
<section>
  <section
    onDoubleClick={[Function]}
  >
    <style
      scoped={true}
    >
      span.Word[data-start="NA"] { background-color: #69e3c2; text-shadow: 0 0 0.01px black }
      span.Word[data-start="NA"]+span { background-color: #69e3c2 }
      span.Word[data-prev-times~="NaN"] { color: #767676 }
      span.Word[data-prev-times~="NaN"] { color: #767676 }
      span.Word[data-confidence="low"] { border-bottom: 1px dotted blue }
    </style>
    <CustomEditor
      customKeyBindingFn={[Function]}
      editorState={
        EditorState {
          "_immutable": Immutable.Record {
            "allowUndo": true,
            "currentContent": Immutable.Record {
              "entityMap": Object {
                "__add": [Function],
                "__create": [Function],
                "__get": [Function],
                "__getLastCreatedEntityKey": [Function],
                "__mergeData": [Function],
                "__replaceData": [Function],
                "add": [Function],
                "create": [Function],
                "get": [Function],
                "getLastCreatedEntityKey": [Function],
                "mergeData": [Function],
                "replaceData": [Function],
              },
              "blockMap": Immutable.OrderedMap {
                "eqvsq": Immutable.Record {
                  "key": "eqvsq",
                  "type": "unstyled",
                  "text": "",
                  "characterList": Immutable.List [],
                  "depth": 0,
                  "data": Immutable.Map {},
                },
              },
              "selectionBefore": Immutable.Record {
                "anchorKey": "eqvsq",
                "anchorOffset": 0,
                "focusKey": "eqvsq",
                "focusOffset": 0,
                "isBackward": false,
                "hasFocus": false,
              },
              "selectionAfter": Immutable.Record {
                "anchorKey": "eqvsq",
                "anchorOffset": 0,
                "focusKey": "eqvsq",
                "focusOffset": 0,
                "isBackward": false,
                "hasFocus": false,
              },
            },
            "decorator": null,
            "directionMap": Immutable.OrderedMap {
              "eqvsq": "LTR",
            },
            "forceSelection": false,
            "inCompositionMode": false,
            "inlineStyleOverride": null,
            "lastChangeType": null,
            "nativelyRenderedContent": null,
            "redoStack": Immutable.Stack [],
            "selection": Immutable.Record {
              "anchorKey": "eqvsq",
              "anchorOffset": 0,
              "focusKey": "eqvsq",
              "focusOffset": 0,
              "isBackward": false,
              "hasFocus": false,
            },
            "treeMap": Immutable.OrderedMap {
              "eqvsq": Immutable.List [
                Immutable.Record {
                  "start": 0,
                  "end": 0,
                  "decoratorKey": null,
                  "leaves": Immutable.List [
                    Immutable.Record {
                      "start": 0,
                      "end": 0,
                    },
                  ],
                },
              ],
            },
            "undoStack": Immutable.Stack [],
          },
        }
      }
      handleKeyCommand={[Function]}
      onChange={[Function]}
      onWordClick={[Function]}
      setEditorNewContentState={[Function]}
      stripPastedStyles={true}
    />
  </section>
</section>
`;
